File: src/backend/TrafficCourts/Messaging/MessageContracts/DisputeStatus.cs
namespace TrafficCourts.Messaging.MessageContracts
{
    /// <summary>
    /// An enumeration of available Statuses on a Dispute record.
    /// </summary>
    public enum DisputeStatus
    {
        /// <summary>
        /// The dispute is new.
        /// </summary>
        NEW,
        
        /// <summary>
        /// The dispute is being processed.
        /// </summary>
        PROCESSING,

        /// <summary>
        /// The dispute was rejected by the VTC.
        /// </summary>
        REJECTED,
        
        /// <summary>
        /// The dispute was cancelled by the disputant.
        /// </summary>
        CANCELLED,

        /// <summary>
        /// The dispute was validated by the VTC.
        /// </summary>
        VALIDATED
    }
}

File: src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/model/DisputeStatus.java
package ca.bc.gov.open.jag.tco.oracledataapi.model;

/**
 * An enumeration of available Statuses on a Dispute record.
 */
public enum DisputeStatus {

	/** Unknown type (undefined). Must be index 0. */
	UNKNOWN("UKN"),
	NEW("NEW"),
	VALIDATED("VALD"),
	PROCESSING("PROC"),
	REJECTED("REJ"),
	CANCELLED("CANC"),
	CONCLUDED("CNLD");

	private String shortName;

	private DisputeStatus(String shortName) {
		this.shortName = shortName;
	}

	public String toShortName() {
		return shortName;
	}

}
